### PR TITLE
Signup page and next auth

### PR DESCRIPTION
**Pull request by davideastmond**
- Created the signup  (registration) form
- Created reusable text fields (for normal text and for password entry)
- Created reusable button components

- Implemented nextAuth. Sessions are created
- Users are inserted into the database via the signup form
- Form fields are validated using `yup`

- Created an "AuthClient" - which groups auth related API functionality into one object.
image

### CODE REVIEW

**Review by AlanCLN — APPROVED** [review]
Looks good to me

File: package.json
Comment (Tandid) [inline package.json:1]: I noticed we added the emotion and material ui packages for the css, but we'll also be using tailwind css. Should we stick with one css framework or use multiple? It's not a huge issue, just wondering if it's ideal to use multiple or stick with one for consistency.

File: src/schemas/user.schema.ts
Comment (Tandid) [inline src/schemas/user.schema.ts:1]: In general the code for this looks good, but we should definitely add more properties to the schema later.

**Review by Tandid — APPROVED** [review]
This was really well done! I tested the code and it seems to be working just fine. I tested the signup and the created user gets updated in my local database and redirects me to the homepage which is great. Also, I like that many of the components are reusable. We should definitely update the UserSchema and decide if we want to use multiple CSS libraries. I'm okay with using multiple, but I'm wondering if it makes the code too messy for others.

**Pull request merged**